rfcs/0177-Skip-ci-integrations.md

```diff
@@ -0,0 +1,34 @@
+# RFC 177 - Skip CI in github integration
+* Comments: [#177](https://github.com/taskcluster/taskcluster-rfcs/pull/177)
+* Proposed by: @lotas
+
+# Summary
+
+Allow to skip running tasks for certain commits containing `[ci skip]`/`[skip ci]` keyword.
+
+## Motivation
+
+This will allow to save computing resources and give more flexibility for the Taskcluster users on Github.
+Not all commits require tasks to be run, for example, updating README.
+All major CI platforms provide this functionality.
+
+# Details
+
+This will be implemented in the webhook API handler. Depending on the `.taskcluster.yml` configuration, tasks could be created for both `push` and `pull-request` events, so we'll have to do those checks on both.
+
+## `push` event
+
+For the push event, we will check if the head commit includes one of the `[ci skip]`, `[skip ci]` keywords in its message. If message is found, this webhook event will be ignored. No messages will be published to the rabbitmq exchange, so no tasks will be started from this event.
+
+If push consists of multiple commits, we would only check the head or the latest one.
+
+## `pull_request` event
+
+To prevent whole pull request from running tasks, we will check if its title contains `[ci skip]`, `[skip ci]` keywords.
```

Comment:
Another options could be:
1. Checking PR's description as well
2. Checking latest commit instead. 

Having `[ci skip]` in PR title imo is better, as you would immediately see in the list of open pull requests which ones will not run jobs. Plus, your commit messages (of the commits that will be pushed to that PR) can stay clean from those keywords and will not appear in history later.

This event doesn't include information about pushed commits, so to keep implementation simple, code will not do extra calls to Github API to retrieve the latest commit. Having PR information should be sufficient to make this decision

Reply:
I like having the `skip` tags in the PR title and/or description for the reasons you just mentioned in this comment. Keeps the commit messages nice and clean.

Reply:
Oh, sorry, I've fallen again in the same trap .. forgot we have both events, so even if your PR includes this marker, `push` events will only check the commit message. I think this argument is invalid then.